content/blog/2025-04-02-introducing-bloom-filters.md

```diff
@@ -0,0 +1,188 @@
++++
+title= "Introducing Bloom Filters for Valkey"
+description = "Learn how to use bloom filters to perform large-scale membership testing with significant memory savings."
+date= 2025-04-02 01:01:01
+authors= [ "karthiksubbarao"]
++++
+
+We are introducing Bloom Filters as a new data type in Valkey via [valkey-bloom](https://github.com/valkey-io/valkey-bloom/), an official Valkey Module which is compatible with Valkey versions >= 8.0. Bloom filters provide efficient, large-scale membership testing, improving performance and offering significant memory savings for high-volume applications.
+
+As an example, previously, Valkey users could use the `SET` data type for handling advertisement deduplication workloads. However, since every item in the set is allocated, the problem with this approach is high memory usage. This article demonstrates how using the bloom filter data type from valkey-bloom achieves >93% memory savings for the same workload, while exploring its implementation, technical details, and practical recommendations.
```

Comment:
Reflow the lines at some point, it makes it much easier to review and leave comments. I like doing it after each sentence, but you can also do it at line lengths.

Reply:
On this line, the use case isn't described very clearly before explaining it uses a lot of memory. Ideally we would fully describe the problem up front, describe the solution, then talk about how it's implemented.

Reply:
> Reflow the lines at some point, it makes it much easier to review and leave comments. I like doing it after each sentence, but you can also do it at line lengths.

Sure, I can move every sentence into a new line

Reply:
> On this line, the use case isn't described very clearly before explaining it uses a lot of memory. Ideally we would fully describe the problem up front, describe the solution, then talk about how it's implemented.

Thank you - I added more context in the intro now. Let me know if I should add anything more.